### DISCUSSION TITLE
Laravel Livewire data is converted to an array after render

### DISCUSSION THREAD

**Discussion by iqlasuddin** [livewire/livewire#4029] [1. Help]
I have a simple livewire component which initializes with queried data from SQL. Below is code

```
public $states=[];

public function mount()
{
    $this->states = DB::select("select id,name from hr_states");
}
```

in the view:

```
<select class="mt-1">
    <option>...select...</option>
    @foreach($states as $state)
        <option value="{{$state->id}}" wire:key="{{$state->id}}">{{$state->name}}</option>
    @endforeach
</select>
```

on the first render, it works well but on any update on page (on other events), it throws error `Attempt to read property "id" on array` pointing to this line` <option value="{{$state->id}}" wire:key="{{$state->id}}">{{$state->name}}`

So I tried rendering as an array `{{$state['id']}}` but that throws error `Cannot use object of type stdClass as array`

**Comment by joshhanley** [comment]
@iqlasuddin Livewire doesn't support re-hydrating raw `DB` calls, as they return `stdClass` objects. What happens is when the `stdClass` object is used on the first request, it is an object hence why you can access properties on it.

To send it to the front end, the objects gets serialised to arrays, but on subsequent requests, Livewire doesn't know how to rehydrate them so just populates it with the data it has, which is the array.

So you are best to actually use queries that return Eloquent models or call `->toArray()` on your `DB` call result and work with the data as arrays.

Hope this helps!

**Reply by ZermattChris** [reply]
Thanks from me too -- quite new to both Laravel & Livewire... All was going well, then suddenly was getting a stdClass object or an Array. Blew up after first render. All moved to Arrays now.
